dtfs-central-api/api-tests/v1/utilisation-reports/put-fee-record-correction-transient-form-data.api-test.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  FEE_RECORD_STATUS,
+  FeeRecordEntityMockBuilder,
+  MAX_RECORD_CORRECTION_ADDITIONAL_INFO_CHARACTER_COUNT,
+  RECONCILIATION_IN_PROGRESS,
+  RECORD_CORRECTION_REASON,
+  RecordCorrectionTransientFormData,
+  UtilisationReportEntityMockBuilder,
+} from '@ukef/dtfs2-common';
+import { withSqlIdPathParameterValidationTests } from '@ukef/dtfs2-common/test-cases-backend';
+import { testApi } from '../../test-api';
+import { SqlDbHelper } from '../../sql-db-helper';
+import { aTfmSessionUser } from '../../../test-helpers';
+import { PutFeeRecordCorrectionTransientFormDataPayload } from '../../../src/v1/routes/middleware/payload-validation';
+
+console.error = jest.fn();
+
+const BASE_URL = '/v1/utilisation-reports/:reportId/fee-records/:feeRecordId/correction-transient-form-data';
+
+describe(`PUT ${BASE_URL}`, () => {
+  const getUrl = (reportId: number | string, feeRecordId: number | string) =>
+    BASE_URL.replace(':reportId', reportId.toString()).replace(':feeRecordId', feeRecordId.toString());
+
+  const reportId = 1;
+  const feeRecordId = 2;
+
+  const aValidRequestBody = () => ({
+    formData: {
+      reasons: [RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT, RECORD_CORRECTION_REASON.OTHER],
+      additionalInfo: 'Some additional info',
+    },
+    user: aTfmSessionUser(),
+  });
+
+  beforeAll(async () => {
+    await SqlDbHelper.initialize();
+
+    await SqlDbHelper.deleteAllEntries('UtilisationReport');
+  });
+
+  beforeEach(async () => {
+    const report = UtilisationReportEntityMockBuilder.forStatus(RECONCILIATION_IN_PROGRESS).withId(reportId).build();
+    const feeRecord = FeeRecordEntityMockBuilder.forReport(report).withId(feeRecordId).withStatus(FEE_RECORD_STATUS.TO_DO).build();
+    report.feeRecords = [feeRecord];
+
+    await SqlDbHelper.saveNewEntry('UtilisationReport', report);
+  });
+
+  afterEach(async () => {
+    await SqlDbHelper.deleteAllEntries('FeeRecordCorrectionTransientFormData');
+    await SqlDbHelper.deleteAllEntries('UtilisationReport');
+  });
+
+  withSqlIdPathParameterValidationTests({
+    baseUrl: BASE_URL,
+    makeRequest: (url) => testApi.put({}).to(url),
+  });
+
+  const requiredPayloadKeys: (keyof PutFeeRecordCorrectionTransientFormDataPayload)[] = ['formData', 'user'];
+  const requiredFormDataPayloadKeys: (keyof RecordCorrectionTransientFormData)[] = ['reasons', 'additionalInfo'];
+
+  it.each([requiredPayloadKeys])(`returns a '${HttpStatusCode.BadRequest}' when the '%s' field is missing`, async (payloadKey) => {
+    // Arrange
+    const requestBody = {
+      ...aValidRequestBody(),
+      [payloadKey]: undefined,
+    };
+
+    // Act
+    const response = await testApi.put(requestBody).to(getUrl(reportId, feeRecordId));
+
+    // Assert
+    expect(response.status).toEqual(HttpStatusCode.BadRequest);
+  });
+
+  it.each(requiredFormDataPayloadKeys)(`returns a '${HttpStatusCode.BadRequest}' when the formData field is missing its '%s' field`, async (payloadKey) => {
```

Comment:
same as above and any other places in this PR

Reply:
Prefixed all it blocks in this PR with 'should'